plugins/rhacm/src/components/ClusterContext/ClusterContext.tsx

```diff
@@ -32,19 +30,25 @@ export const ClusterContextProvider = (props: any) => {
     { loading: true },
   );
   useDebounce(refresh, 10);
-
-  let errorBool = false;
-  /*
-  Error can either be undefined or Error(), if its undefined, no error happended
-  if cluster has some 'error' key, it means the API sent back an error response
-  */
-  if (error instanceof Error || 'error' in cluster) {
-    errorBool = true;
-  }
+  const isError = Boolean(error || 'error' in cluster);
+  const errorObject = isError
+    ? error ||
+      Object.assign(
+        new Error((cluster as unknown as ErrorResponseBody)?.error?.message),
+        {
+          ...(cluster as unknown as ErrorResponseBody)?.error,
+        },
+      )
+    : null;
 
   return (
     <ClusterContext.Provider
-      value={{ data: cluster, loading, error: errorBool }}
+      value={{
+        data: isError || loading ? null : cluster,
+        loading,
+        isError,
+        errorObject,
```

Comment:
why is this information exported twice? Isn't in this case `isError === Boolean(errorObject)`?

I don't think there's any benefit in exporting `isError`, `errorObject` (better named `error` is enough).

See implementation of for example `react-firebase-hooks` as a reference:
https://www.npmjs.com/package/react-firebase-hooks
https://github.com/CSFrequency/react-firebase-hooks/tree/master/firestore